src/Connector.js

```diff
@@ -221,18 +221,16 @@ async function removeDeviceFromOrion(orionUrl, id) {
 
   const promises = sensors.map(async (sensor) => {
     url = `${orionUrl}/v2/entities/${sensor.id}`;
-    await request.delete({ url, headers, json: true });
+    return request.delete({ url, headers, json: true });
   });
 
   await Promise.all(promises);
 }
 
 async function subscribeToEntities(client, devices) {
   const promises = devices.map(async (device) => {
-    await client.subscribe(`/default/${device.id}/cmd`);
-    device.schema.map(async (sensor) => {
-      await client.subscribe(`/${device.id}/${sensor.sensorId}/cmd`);
-    });
+    client.subscribe(`/default/${device.id}/cmd`);
+    return device.schema.map(async sensor => client.subscribe(`/${device.id}/${sensor.sensorId}/cmd`));
   });
 
   await Promise.all(promises);
```

Comment:
The `promises` will have just the subscribes about the sensor and not from the devices.

Reply:
Totally agree, makes sense! :smile: